hermes/probe/delete/delete_file.go

```diff
@@ -0,0 +1,138 @@
+// Returns:
+//	- fileID: returns the ID of the file delete OR a missing file to be created if one is found.
+//	- err:
+//		- [...].hermes_file_missing: the file to be deleted does not exist in Hermes' StateJournal.
+//		- [...].file_missing: the file to be deleted could not be found in the target bucket.
+//		- [...].bucket_missing: the target bucket on this target system was not found.
+//		- [...].probe_failed: there was an error during one of the API calls and the probe failed.
+//		- [...].deleted_file_found: the file was deleted but it is still found in the target bucket.
+//		- [...].list_bucket_failed: the listBucket operation failed when checking if the target file was deleted.
+func DeleteFile(ctx context.Context, fileID int32, target *probe.Target, client *stiface.Client, logger *logger.Logger) (int32, error) {
+	bucket := target.Target.GetBucketName()
+
+	filename, ok := target.Journal.Filenames[fileID]
+	if !ok {
+		return fileID, fmt.Errorf("deleteRandomFile(id: %d).hermes_file_missing: could not delete file %s, file not found", fileID, filename)
+	}
+
+	file := (*client).Bucket(bucket).Object(filename)
+
+	start := time.Now()
+	if err := file.Delete(ctx); err != nil {
+		var status m.ExitStatus
+		switch err {
+		case storage.ErrObjectNotExist:
+			status = m.FileMissing
+		case storage.ErrBucketNotExist:
+			status = m.BucketMissing
+		default:
+			status = m.ProbeFailed
+		}
+
+		target.LatencyMetrics.APICallLatency[m.APIDeleteFile][status].Metric("latency").AddFloat64(time.Now().Sub(start).Seconds())
+		return fileID, fmt.Errorf("deleteRandomFile(id: %d).%v: could not delete file %s: %w", fileID, status, filename, err)
+	}
+	target.LatencyMetrics.APICallLatency[m.APIDeleteFile][m.Success].Metric("latency").AddFloat64(time.Now().Sub(start).Seconds())
+
+	query := &storage.Query{Prefix: filename}
+	start = time.Now()
+	objects := (*client).Bucket(bucket).Objects(ctx, query)
+	target.LatencyMetrics.APICallLatency[m.APIListFiles][m.Success].Metric("latency").AddFloat64(time.Now().Sub(start).Seconds())
+	for {
+		obj, err := objects.Next()
+		if obj.Name == filename {
+			return fileID, fmt.Errorf("deleteRandomFile(id %d).deleted_file_found: object %v in bucket %q still listed after delete", fileID, *obj, bucket)
+		}
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return fileID, fmt.Errorf("deleteRandomFile(id: %d).list_bucket_failed: unable to list bucket %q: %w", fileID, bucket, err)
+		}
+	}
+
+	// Update in-memory NIL file after delete operation.
+	delete(target.Journal.Filenames, int32(fileID))
+
+	logger.Infof("Object %v deleted in bucket %s.", file, bucket)
+	return fileID, nil
+}
+
+// pickFileToDelete picks which file to delete and returns the integer ID of this file.
+// Returns:
+//	- ID: returns the ID of the file to be deleted.
+func pickFileToDelete() int32 {
+	const (
+		beg                    = 10 // we can delete files staring from the file Hermes_10
```

Comment:
change beg to begin (more descriptive)
also since we changed our indexing (index from 0 but 0 is nil file) and we keep files 1-10 for persistence (right?) then we want to delete files from 11 to 50 included so begin = 11 and numberOfDeletableFiles = 40

Reply:
Ack.